addons/base_address_extended/__init__.py

```diff
@@ -7,4 +7,10 @@
 
 def _update_street_format(cr, registry):
     env = api.Environment(cr, SUPERUSER_ID, {})
-    env['res.partner'].search([])._compute_street_data()
+    partners = env['res.partner'].search([])
+    if len(partners) < 20000:
+        partners._compute_street_data()
+    else:
+        # For large DB, a perfect data that crashes is worse than an unperfect one that does not crash.
```

Comment:
Hi, for cases similar to this, adding an async worker (similar to [oca's queue_job](https://github.com/OCA/queue)) would solves a lot of problems.

Reply:
I like this idea

Reply:
For now this will work - I will look at a cron in a separate PR.
Neither this method nor cron will impact the database load time since New dbs won't have any partners.
The cron will just be 'nice to have'

Reply:
@smetl you were right - `compute` and `inverse` are both called, however theres something more interesting...
On module installation, after creating the database tables, the stored computed fields are prepared ... so `_compute_street_data` is executed once already.
*After* that, the post_init_hook is called and `_update_street_format` does this for a second time. And this time it executes both compute and inverse...

So in an attempt to gain performance, we are actually wasting it :)

Reply:
removed the post_init_hook since it is doing a computation that is already done by the ORM for stored, computed fields